apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
+
+	lib_memsostream((FAR struct lib_memsostream_s *)&memsostream, buf + 4, BUFLEN);
+	TC_ASSERT_EQ("lib_memsostream", (memsostream.buffer - (FAR char *)(buf)), 4);
+	TC_ASSERT_EQ("lib_memsostream", memsostream.buflen, (BUFLEN - 1));	/* Save space for null terminator, hence checing with (BUFLEN-1)*/
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_nullinstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will return only EOF
+* @scenario         Initializes a NULL stream. The initialized stream will return only EOF
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+        union {
+                struct lib_instream_s nullinstream;
+                struct lib_meminstream_s meminstream;
+        } instream_object;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_instream_s *stream;
+
+	lib_meminstream((FAR struct lib_meminstream_s *)&instream_object.meminstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_meminstream", instream_object.meminstream.buffer, (FAR char *)(buf));
+
+	lib_nullinstream(&instream_object.nullinstream);
+	stream = &instream_object.nullinstream;
+	TC_ASSERT_EQ("lib_nullinstream", instream_object.nullinstream.nget, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_nulloutstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @scenario         Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @apicovered       lib_nulloutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nulloutstream_tc(void)
+{
+        union {
+                struct lib_outstream_s nulloutstream;
+                struct lib_memoutstream_s memoutstream;
+        } outstream_object;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_outstream_s *stream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&outstream_object.memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", outstream_object.memoutstream.buffer, (FAR char *)(buf));
+
+	lib_nulloutstream(&outstream_object.nulloutstream);
+	stream = &outstream_object.nulloutstream;
+	TC_ASSERT_EQ("lib_nulloutstream", outstream_object.nulloutstream.nput, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_rawinstream_tc
+* @brief            Initializes a stream for use with a file descriptor
+* @scenario         Initializes a stream for use with a file descriptor
+* @apicovered       lib_rawinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT_NEQ("fopen", fd, OK);
+
+	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, (int)fd);
```

Comment:
You got the file pointer and changed type of it to get fd value.
Is it correct? It looks wrong.
We don't need to get fd using open?